lib/common.js

```diff
@@ -937,6 +937,47 @@ function getServerNicTags(cb) {
 	});
 }
 
+/*
+ * Query SAPI for the current update_channel. The callback function takes
+ * the form function(err, update_channel)
+ * This function can have a SAPI and bunyan log object passed as options,
+ * or will fallback to objects initialized by initSdcClients and self.log
+ * if those are not provided.
+ * The update channel can be empty, (e.g. if `sdcadm channel unset`)
+ * in which case we set return a null.
+ */
+function getSdcChannel(opts, cb) {
+	assert.object(opts, 'opts');
+	assert.object(opts.sapi, 'opts.sapi');
+	assert.object(opts.log, 'opts.log');
+	assert.func(cb, 'cb');
+
+	var sapi = opts.sapi;
+	var log = opts.log;
+
+	var search = {
+		name: 'sdc',
+		include_master: true
+	};
+
+	sapi.listApplications(search, function (err, apps) {
+		if (err) {
+			log.error(err, 'failed to list applications');
+			return (cb(err));
+		}
+
+		log.debug('found sdc application');
+		assert.ok(apps.length <= 1);
```

Comment:
probably want to check that it is exactly 1. If it is zero, then your next `apps[0]` is going to break. However if it is 0 then no way you got to this point. ;)

Reply:
Ha, good point. I'll change that just to make it very clear.